compiler/types.nim

```diff
@@ -1280,6 +1280,8 @@ proc commonSuperclass*(a, b: PType): PType =
   var ancestors = initIntSet()
   while x != nil:
     x = skipTypes(x, skipPtrs)
+    if skipRoot and sfCompilerProc in x.sym.flags:
+      break
```

Comment:
What does this really skip here?

Reply:
skip Root objects and Exception they are compiler defined type

Reply:
Why would these be special cases?

Reply:
make RootObj the common type, then next don't match the error message will be expect Root Obj but get X, sure A B inherit from RootObj directly they are common in RootObj, but no relationship in typeRel

Reply:
Hmm ok.